src/metabase/server/middleware/security.clj

```diff
@@ -124,11 +124,52 @@
           "Content-Security-Policy"
           #(format "%s frame-ancestors %s;" % (if allow-iframes? "*" (or (embedding-app-origin) "'none'")))))
 
+(defn- parse-url [url]
+  (let [pattern #"^(?:(https?)://)?([^:/]+)(?::(\d+|\*))?$"
+        matches (re-matches pattern url)]
+    (if matches
+      (let [[_ protocol domain port] matches]
+        {:protocol protocol
+         :domain domain
+         :port port})
+      (throw (IllegalArgumentException. "Invalid URL")))))
+
+(defn approved-origin?
+  "Returns true if `origin` should be allowed for CORS based on the `approved-origins`"
+  [raw-origin approved-origins]
+  (if (or (empty? raw-origin) (empty? approved-origins))
+    false
+    (let [approved-list (str/split approved-origins #" ")
+          origin        (parse-url raw-origin)]
+      (boolean (some (fn [approved-origin-raw]
```

Comment:
It's best practice to use when instead of an if with a nil branch. Since we want to return a boolean, wrap the entire function body with the boolean instead.

Something like:

```clojure
(boolean
  (when (...)
    (let [approved-list (str/split approved-origins #" ")
          origin        (parse-url raw-origin)]
      (boolean (some (fn [approved-origin-raw]))))
```